# INFERENCIA ESTADISTICA: Segunda Práctica

## Estimaciones e Intervalos de Confianza

En esta practica vamos a ver con qué comandos de R se calculan intervalos de confianza para una y dos poblaciones. Usaremos el fichero alturas.txt que usamos en la primera sesion para aprender 
estos comandos

In [29]:
Datos <-read.table(file="alturas.txt", header=TRUE, dec=",", sep="\t")




### Peso medio de la poblacion: estimacion puntual e Intervalo de Confianza

In [8]:
t.test(Datos$PESO,alternative='two.sided',conf.level=.95)


	One Sample t-test

data:  Datos$PESO
t = 52.957, df = 170, p-value < 2.2e-16
alternative hypothesis: true mean is not equal to 0
95 percent confidence interval:
 69.83973 75.24799
sample estimates:
mean of x 
 72.54386 


#### Explicacion
El valor de la estimacion puntual es 72.54386 y el intervalo de confianza con nivel del confianza del 95% es (69.83973,75.24799)

#### Nota: Otro nivel de confianza
Para considerar otro nivel del confianza, hay que cambiar conf.level por el valor a considerar. Por ejemplo, el intervalo de confianza del 99% hay que cambiar conf.level=.95 por conf.level=.99

#### Nota: Otra variable
Para considerar otra variable, hay que cambiar el primer parámetro del comando. Por ejemplo, el intervalo de confianza del 95% de la altura media seria 

t.test(Datos$ALTURA,alternative='two.sided',conf.level=.95)


### Varianza de la altura de la población: estimacion puntual e Intervalo de Confianza del 95%

In [14]:
# Quizas, hay que instalar y cargar el siguiente paquete  
# install.packages("EnvStats")
 library(EnvStats)
varTest(Datos$ALTURA,alternative  = "two.sided",conf.level=0.95)


	Chi-Squared Test on Variance

data:  Datos$ALTURA
Chi-Squared = 1884, df = 170, p-value < 2.2e-16
alternative hypothesis: true variance is not equal to 1
95 percent confidence interval:
  9.057947 13.874455
sample estimates:
variance 
11.08242 


#### Explicacion:
El valor de la estimacion puntual de la varianza de la poblacion es de 11.08242 y el intervalo de confianza del 95% es (0.353491, 13.376088)

#### Nota: Intervalo de Confianza de la desviación estandar de la poblacion
Si en vez de la varianza, queremos considerar la desviación estandar, se usa la funcion sqrt. Para nuestro ejemplo, seria lo siguiente

In [15]:
# Estimacion puntual de la desviacion estandar.
sqrt(11.08242)
# Valor inferior del intervalo de confianza 
sqrt(9.353491)
# Valor superior del intervalo de confianza
sqrt(13.376088)

[1] 3.329027

[1] 3.058348

[1] 3.657333

## Cociente de Varianzas de dos poblaciones independientes

### Estimacion del cociente de varianzas de alturas de hombres y mujeres

In [16]:
mujeres <-Datos$ALTURA[Datos$SEXO=="mujer"]       
hombres <-Datos$ALTURA[Datos$SEXO=="hombre"]
var.test(mujeres,hombres,conf.level=0.95)


	F test to compare two variances

data:  mujeres and hombres
F = 0.88431, num df = 81, denom df = 88, p-value = 0.576
alternative hypothesis: true ratio of variances is not equal to 1
90 percent confidence interval:
 0.6177924 1.2697464
sample estimates:
ratio of variances 
         0.8843092 


#### Explicacion:
El valor estimado para el cociente de las varianzas de las alturas entre hombres y mujeres es 0.8843092 y el intervalo de confianza del %95eko es(0.6177924, 1.2697464). Por tanto, no podemos concluir que las varianzas sean distintas (ya que el 1 está dentro del intervalo).

#### Nota: Desviación estandar
Si en vez de la varianza, queremos considerar la desviación estandar, se usa la funcion sqrt de la misma forma que se ha explicado arriba.

## Diferencia de medias de dos poblaciones independientes

### Intervalo de Confianza del 90% de la diferencia de la altura media entre hombres y mujeres

In [18]:
# En el anterior apartado, hemos visto que no podemos rechazar la igualdad de varianzas, 
# Por eso, ponemos var.equal=TRUE en el comando de abajo (si las varianzas fuese distintas
#   entonces  var.equal=FALSE es lo que hay que poner)
t.test(mujeres,hombres,conf.level=0.9,var.equal=TRUE)


	Two Sample t-test

data:  mujeres and hombres
t = -4.0665, df = 169, p-value = 7.304e-05
alternative hypothesis: true difference in means is not equal to 0
90 percent confidence interval:
 -2.790305 -1.176809
sample estimates:
mean of x mean of y 
 165.7805  167.7640 


#### Explicacion:
Estimacion puntual de la altura de las mujeres: 165.7805

Estimacion puntual de la altura de los hombres: 167.7640

Intervalo de Confianza del 90% de la diferencia de la altura media de hombres y mujeres: (-2.790305,-1.176809)

#### Conclusion:
Los valores obtenidos en el intervalo de confianza son negativos, por tanto concluimos con un nivel de confianza del 90% que mu1-mu2 (es decir, la altura media de las mujeres menos la altura media de los hombres) es negativo y, por consiguiente, concluismo que la altura media de las mujeres es menor que la de los hombres.

#### Nota: 
Si en vez de t.test(mujeres,hombres,....) hubiesemos escrito t.test(hombres,mujeres,...) los valores de las estimaciones puntuales no variarian, pero el intervalo de confianza tendría valores positivos. Por tanto, obtendriamos las mismas conclusiones.

## Datos Pareados

### Intervalo de Confianza del 98% de la diferencia entre la altura del padre y la madre

In [19]:
t.test(Datos$PADRE,Datos$MADRE, conf.level=0.98)


	Welch Two Sample t-test

data:  Datos$PADRE and Datos$MADRE
t = 23.914, df = 316.34, p-value < 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
98 percent confidence interval:
 10.19882 12.40937
sample estimates:
mean of x mean of y 
 172.1871  160.8830 


#### Explicacion:
Estimacion puntual de la altura de las madres: 172.1871

Estimacion puntual de la altura de los padres: 160.8830

Intervalo de Confianza del 98% de la diferencia de alturas de los padres y las madres (10.19882,12.40937)

#### Conclusion:
Los valores obtenidos en el intervalo de confianza son positivos, por tanto concluimos con un nivel de confianza del 98% que muD (es decir, la altura media de los padres menos la altura media de las madres) es positivo y, por consiguiente, concluismo que la altura media de las madres es menor que la de los padres.



## Parámetro p de una poblacion Bin(1,p)

### Intervalo de Confianza del 95% de los que son de Alava

In [24]:
# contamos cuantos son de Alava
Datos.alava <- subset(Datos,PROVINCIA==1)
cuantos.alava <- length(Datos.alava$ALTURA)
# ahora contamos cuantos hay en total
cuantos.total <- length(Datos$ALTURA)

prop.test(cuantos.alava,cuantos.total,conf.level=0.95)


	1-sample proportions test with continuity correction

data:  cuantos.alava out of cuantos.total, null probability 0.5
X-squared = 49.497, df = 1, p-value = 1.987e-12
alternative hypothesis: true p is not equal to 0.5
95 percent confidence interval:
 0.1689802 0.2996563
sample estimates:
        p 
0.2280702 


#### Explicacion:
Estimacion puntual de la proporcion de los que son de Alava: 0.2280702

Intervalo de Confianza del 95%: (0.1689802,0.2996563)



## Dos poblaciones Bin(1,p1) y Bin(1,p2)

### Diferencia de proporcion de los que son de Alava entre hombres y mujeres

In [33]:
# contamos cuantas mujeres son de Alava y cuantos hombres son de Alava
Datos.alava <- subset(Datos,PROVINCIA==1)
mujer.alava <- subset(Datos.alava,SEXO=="mujer")
hombre.alava <- subset(Datos.alava,SEXO=="hombre")
n_mujer_alava <- length(mujer.alava$ALTURA)
n_hombre_alava <- length(hombre.alava$ALTURA)
# contamos cuantos hombres y mujeres hay en total
Datos.mujeres <- subset(Datos,SEXO=="mujer")
n_mujeres <- length(Datos.mujeres$ALTURA)
Datos.hombres <- subset(Datos,SEXO=="hombre")
n_hombres <- length(Datos.hombres$ALTURA)
# hacemos el test
prop.test(c(n_mujer_alava,n_hombre_alava),c(n_mujeres,n_hombres),conf.level=0.95)



	2-sample test for equality of proportions with continuity correction

data:  c(n_mujer_alava, n_hombre_alava) out of c(n_mujeres, n_hombres)
X-squared = 0.43037, df = 1, p-value = 0.5118
alternative hypothesis: two.sided
95 percent confidence interval:
 -0.08391627  0.19161701
sample estimates:
   prop 1    prop 2 
0.2560976 0.2022472 


#### Explicacion:
La estimacion puntual de mujeres que son de Alava: 0.2560976

La estimacion puntual de mujeres que son de Alava: 0.2022472

Intervalo de Confianza del 95% de la diferencia de la proporcion de mujeres de Alava y hombres de Alava: (-0.08391627,0.19161701)

#### Conclusion:
Como el zero está dentro del intervalo, no podemos concluir que haya una desigualdad en la proporcion de los que son de Alava entre mujeres y hombres.
